### PR TITLE
Project process md5

### LINKED ISSUE #267: Project process data output is missing MD5s
`projects.Output` is missing MD5 checking. There's also some code duplication between this and sample output. #262 will make `transfer_data.create_output_links` generic, so should be able to use that.

### PR DESCRIPTION
**Pull request by katieemelianova**
run md5sum after project process
closes #267

### CODE REVIEW

File: etc/example_analysisdriver.yaml [hunk 1]
```diff
@@ -83,3 +92,8 @@ default:
 
     well_duplicate:
       coord_file: path/to/coord_file
+
+    references:
+        Homo sapiens:
+            fasta: /lustre/home/production/Resources/genomes/Homo_sapiens/hg38.fa
+            dbsnp: /lustre/home/production/Applications/bcbio/0.9.8/genomes/Hsapiens/hg38/variation/dbsnp-147.vcf.gz
```
Comment (mwhamgenomics) [inline etc/example_analysisdriver.yaml]: The 'references' config field was removed in #221. This should be:
```yaml
species:
    'Homo sapiens':
        default: a_genome

genomes:
    a_genome:
        fasta: path/to/genome.fa
        dbsnp: path/to/dbsnp.vcf.gz
```
`projects.build_pipeline` should then query this via `dataset.reference_genome`

File: tests/test_pipelines/test_projects.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    def test_run(self, mocked_execute):
+        with patch('analysis_driver.segmentation.BasicStage.job_dir', new=os.path.join(self.assets_path, 'test_projects')):
+            self.md5._run()
+
+
+        mocked_execute.assert_called_once_with(cpus=1,
+                                               job_name='md5sum',
+                                               log_commands=False,
+                                               mem=2,
+                                               working_dir='/Users/katie/PycharmProjects/Analysis-Driver/tests/assets/test_projects')
```
Comment (mwhamgenomics) [inline tests/test_pipelines/test_projects.py]: This is breaking the build.

File: analysis_driver/pipelines/projects.py [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
 from analysis_driver.exceptions import PipelineError
-from analysis_driver.transfer_data import output_data_and_archive
+from analysis_driver.transfer_data import output_data_and_archive, create_output_links
 
 toolset_type = 'project_processing'
 name = 'project'
 
 
+def delivery_source():
+    return cfg.query('sample', 'delivery_source')
```
Comment (mwhamgenomics) [inline analysis_driver/pipelines/projects.py]: Looks like this is only called in one place - does it really need to be a function?

File: analysis_driver/pipelines/projects.py [hunk 4]
```diff
@@ -28,35 +32,39 @@ def build_pipeline(dataset):
...
-                if os.path.islink(symlink_path):
-                    os.unlink(symlink_path)
-                os.symlink(source, symlink_path)
-            else:
-                raise PipelineError('Could not find the file ' + source + ', unable to create link')
+        create_output_links(
+            self.job_dir,
+            output_cfg,
+            dir_with_linked_files
+        )
```
Comment (mwhamgenomics) [inline analysis_driver/pipelines/projects.py:63]: Also missing `toolset.write_to_yaml` for dependency version reporting. I'm feeling like version reporting and md5s should be a standard part of any data output, so there should be a generic DataOutput class in `pipelines.common` that puts all this alongside link creation and the call to output_data_and_archive.

**Review by mwhamgenomics — CHANGES_REQUESTED** [review]

File: analysis_driver/pipelines/projects.py (hunk 4 above)
Comment (katieemelianova) [reply analysis_driver/pipelines/projects.py:63]: I've added toolset.write_to_yaml though have left writing a generic DataOutput for another pull request

**Review by mwhamgenomics — APPROVED** [review]

**Pull request merged**